lib/charms/opensearch/v0/helper_secrets.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Secrets (passwords, certificates, etc.) data handling."""
+
+import logging
+import time
+from typing import Dict, Optional, Union
+
+from charms.opensearch.v0.constants_secrets import (
+    SECRET_LABEL_CERT,
+    SECRET_LABEL_PREFIX,
+    SECRET_LABEL_SEPARATOR,
+)
+from charms.opensearch.v0.constants_tls import CertType
+from charms.opensearch.v0.helper_databag import RelationDataStore, Scope
+from charms.opensearch.v0.opensearch_exceptions import OpenSearchSecretNotFoundError
+from ops import JujuVersion, Secret, SecretNotFoundError
+from overrides import override
+
+# The unique Charmhub library identifier, never change it
+LIBID = "e28df77e11504aef9a537b351fd4cf37"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+
+class SecretsDataStore(RelationDataStore):
```

Comment:
Grrrr, from the live chat I thought it was `opensearch_data` (and I saw the comment much later :-( ).

Should I do the whole renaming again, or it could stay as it is (at least for the sake of another PR? :) )

Reply:
Done.

Reply:
Done.